airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/errors/messages/MongoDbErrorMessage.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.base.errors.messages;
+
+import static io.airbyte.integrations.base.errors.utils.ConnectionErrorType.INCORRECT_ACCESS_PERMISSION;
+import static io.airbyte.integrations.base.errors.utils.ConnectionErrorType.INCORRECT_CLUSTER;
+import static io.airbyte.integrations.base.errors.utils.ConnectionErrorType.INCORRECT_HOST_OR_PORT;
+import static io.airbyte.integrations.base.errors.utils.ConnectionErrorType.INCORRECT_USERNAME_OR_PASSWORD_OR_DATABASE;
+import static io.airbyte.integrations.base.errors.utils.ConnectorName.MONGO;
+
+import io.airbyte.integrations.base.errors.utils.ConnectorName;
+
+public class MongoDbErrorMessage extends ErrorMessage {
```

Comment:
Please add a comment linking to MongoDB doc page that explains these error codes

Reply:
@grishick I reworked exception handling for mongodb. The error code and error message will now be passed directly from mongo exceptions. As for the `-3 and -4` codes - I didn't find them in the documentation, but I see that these numbers are assigned as an error code in the MongoException class in Mongo driver. I am attaching a screenshot where it occurs.
![2022-08-25_15-32](https://user-images.githubusercontent.com/88329385/186665596-628a9932-9c54-40f4-9c03-59f8d591c73a.png)

Reply:
This is helpful to know though. I'm curious to ask, how useful would exposing `state code` if it seems like the `state codes` are arbitrarily assigned with no real perceptible value to surface them?